tests/components/ecobee/test_climate.py

```diff
@@ -1,288 +1,585 @@
+                {"name": "Away", "climateRef": "away"},
+                {"name": "Sleep", "climateRef": "sleep"},
+            ],
+            "currentClimateRef": "home",
+        },
+        "equipmentStatus": "",
+    },
+    {
+        "identifier": "test-identifier-4",
+        "name": "Test Thermostat 4",
+        "modelNumber": "athenaSmart",
+        "settings": {
+            "hvacMode": "cool",
+            "coolStages": 0,
+            "heatStages": 0,
+            "fanMinOnTime": 5,
+            "heatCoolMinDelta": 50,
+            "holdAction": "indefinite",
+        },
+        "runtime": {
+            "connected": True,
+            "actualTemperature": 727,
+            "actualHumidity": 35,
+            "desiredHeat": 753,
+            "desiredCool": 734,
+            "desiredFanMode": "auto",
+        },
+        "events": [
+            {"type": "hold", "name": "Home", "running": True, "holdClimateRef": "home"},
+            {
+                "type": "template",
+                "name": "_Default_",
+                "running": False,
+                "holdClimateRef": "",
+            },
+        ],
+        "program": {
+            "climates": [
+                {"name": "Home", "climateRef": "home"},
+                {"name": "Away", "climateRef": "away"},
+                {"name": "Sleep", "climateRef": "sleep"},
+            ],
+            "currentClimateRef": "home",
+        },
+        "equipmentStatus": "humidifier",
+    },
+    {
+        "identifier": "test-identifier-5",
+        "name": "Test Thermostat 5",
+        "modelNumber": "athenaSmart",
+        "settings": {
+            "hvacMode": "cool",
+            "coolStages": 0,
+            "heatStages": 0,
+            "fanMinOnTime": 5,
+            "heatCoolMinDelta": 50,
+            "holdAction": "nextTransition",
+        },
+        "runtime": {
+            "connected": True,
+            "actualTemperature": 727,
+            "actualHumidity": 35,
+            "desiredHeat": 753,
+            "desiredCool": 734,
+            "desiredFanMode": "auto",
+        },
+        "events": [
+            {
+                "type": "sensor",
+                "name": "Living Room",
+                "running": True,
+                "holdClimateRef": "",
+            },
+            {
+                "type": "template",
+                "name": "_Default_",
+                "running": False,
+                "holdClimateRef": "",
+            },
+        ],
+        "program": {
+            "climates": [
+                {"name": "Home", "climateRef": "home"},
+                {"name": "Away", "climateRef": "away"},
+                {"name": "Sleep", "climateRef": "sleep"},
+            ],
+            "currentClimateRef": "sleep",
+        },
+        "equipmentStatus": "auxHeat2,fan",
+    },
+]
+
+
+@pytest.fixture
+def mock_account(hass):
+    """Mock an ecobee account object."""
+    account = ecobee.EcobeeData(
```

Comment:
We don't want to mock our own code. We should just patch the library interfaces and then set up the integration via core as normal. So don't include integration code in this fixture. It should patch and spec the ecobee library interface and yield within the patch context manager to keep the patch in place during the test.